repos/system_upgrade/el8toel9/actors/dotnet/actor.py

```diff
@@ -0,0 +1,35 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.models import Report, InstalledRPM
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+UNSUPPORTED_VERSIONS = [ '2.1', '3.0', '3.1', '5.0' ]
+
+class DotnetUnsupportedVersionsCheck(Actor):
+    """
+    Check for installed .NET versions that are no longer supported.
+    """
+
+    name = 'dotnet_unsupported_versions_check'
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        unsupported_versions_report_text = ''
+
+        for unsupported_version in UNSUPPORTED_VERSIONS:
+            runtime_package=f'dotnet-runtime-{unsupported_version}'
+            if has_package(InstalledRPM, runtime_package):
```

Comment:
the `InstaledRPM` cannot be used when the model is not in the tuple of consumed models (above, the `consumes` attribute is not even specified, which should be always ideally, even when empty)